environment/src/test/java/com/sequenceiq/environment/environment/service/EnvironmentModificationServiceTest.java

```diff
@@ -688,6 +698,52 @@ void testUpdateAzureResourceEncryptionParametersByEnvironmentCrn() {
 
     }
 
+    @Test
+    void testUpdateAwsDiskEncryptionParametersByEnvironmentName() {
+        UpdateAwsDiskEncryptionParametersDto updateAwsDiskEncryptionParametersDto = UpdateAwsDiskEncryptionParametersDto.builder()
+                .withAwsDiskEncryptionParametersDto(AwsDiskEncryptionParametersDto.builder()
+                        .withEncryptionKeyArn("dummyKeyArn")
+                        .build())
+                .build();
+        Environment env = new Environment();
+        env.setParameters(new AwsParameters());
+        when(environmentService.getValidatorService()).thenReturn(validatorService);
+        when(environmentService
+                .findByNameAndAccountIdAndArchivedIsFalse(eq(ENVIRONMENT_NAME), eq(ACCOUNT_ID))).thenReturn(Optional.of(env));
+        when(validatorService.validateEncryptionKeyArn(any(String.class), any(String.class))).thenReturn(ValidationResult.builder().build());
+        when(environmentDtoConverter.environmentToDto(env)).thenReturn(new EnvironmentDto());
+
+        environmentModificationServiceUnderTest.updateAwsDiskEncryptionParametersByEnvironmentName(ACCOUNT_ID,
+                ENVIRONMENT_NAME, updateAwsDiskEncryptionParametersDto);
+
+        ArgumentCaptor<AwsParameters> awsParametersArgumentCaptor = ArgumentCaptor.forClass(AwsParameters.class);
+        verify(awsParametersRepository).save(awsParametersArgumentCaptor.capture());
+        assertEquals("dummyKeyArn", awsParametersArgumentCaptor.getValue().getEncryptionKeyArn());
+    }
+
+    @Test
+    void testUpdateAwsDiskEncryptionParametersByEnvironmentCrn() {
+        UpdateAwsDiskEncryptionParametersDto updateAwsDiskEncryptionParametersDto = UpdateAwsDiskEncryptionParametersDto.builder()
+                .withAwsDiskEncryptionParametersDto(AwsDiskEncryptionParametersDto.builder()
+                        .withEncryptionKeyArn("dummyKeyArn")
+                        .build())
+                .build();
+        Environment env = new Environment();
+        env.setParameters(new AwsParameters());
+        when(environmentService.getValidatorService()).thenReturn(validatorService);
+        when(environmentService
+                .findByResourceCrnAndAccountIdAndArchivedIsFalse(eq(ENVIRONMENT_NAME), eq(ACCOUNT_ID))).thenReturn(Optional.of(env));
```

Comment:
While it does not make much difference, `ENVIRONMENT_NAME` should be replaced with `CRN`.

Reply:
The file EnvironmentTestData.java does not have CRN. So We should use ENVIRONMENT_NAME only here.

Reply:
So it is time to introduce it! :smile: Seriously, there is `TestConstants.CRN`, or you could add a new constant in this test class (with a value separate from that of `ENVIRONMENT_NAME`, so we can be sure that the right thing can be expected).